Sources/SmithyCodegenCore/AST/ASTError.swift

```diff
@@ -0,0 +1,14 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
```

Comment:
This error just get thrown with an appropriate message if there is an error while reading an AST.